lib/dictBuilder/cover.c

```diff
@@ -620,14 +662,15 @@ static size_t COVER_ctx_init(COVER_ctx_t *ctx, const void *samplesBuffer,
     for (i = 0; i < ctx->suffixSize; ++i) {
       ctx->suffix[i] = i;
     }
-    /* qsort doesn't take an opaque pointer, so pass as a global.
-     * On OpenBSD qsort() is not guaranteed to be stable, their mergesort() is.
-     */
+#if defined(_GNU_SOURCE) || defined(__APPLE__) || defined(_WIN32)
+    unify_qsort_r_interface(ctx);
+#elif defined(__OpenBSD__)
     g_coverCtx = ctx;
-#if defined(__OpenBSD__)
     mergesort(ctx->suffix, ctx->suffixSize, sizeof(U32),
```

Comment:
Do we actually have real data showing the benefits for this special case?

Reply:
This change was implemented in #1100 following a bug report in #1088 .
Possibly an issue with reproducibility in some test scenarios. I don't remember all the details, maybe @terrelln can chime in to explain what sort stability is and why it matters.